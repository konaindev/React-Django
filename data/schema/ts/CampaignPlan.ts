import * as t from "./types";
import { InvestmentExpenses } from "./BaselineReport";

interface TargetSegment {
  /** A description of ordered importance ('primary', 'tertiary') */
  ordinal: string;

  /** A description of the segment ('Young Professionals') */
  description: string;
}

interface Objective {
  /** The primary title for the objective ('Reputation Building') */
  title: string;

  /** A detailed description, in markdown */
  description: t.markdown;
}

interface Overview {
  /** The overall campaign theme ('Variable by audience and segment') */
  theme: string;

  /** The target segments, in order from most important to least */
  target_segments: TargetSegment[];

  /** The overall campaign goal ('achieve lease-up') */
  goal: string;

  /** Campaign objectives */
  objectives: Objective[];

  /** Campaign assumptions */
  assumptions: t.markdown;

  /** Description of the schedule ('Begins in late may 2019') */
  schedule: string;

  /** Target investment, total */
  target: t.currency;

  /** Target investment, per-tactic */
  target_investment: InvestmentExpenses;
}

export interface CampaignPlan {
  overview: Overview;
}
