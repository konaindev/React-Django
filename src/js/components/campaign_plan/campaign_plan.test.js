import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import CampaignPlan from "./index";
import CampaignPlanProps from "./campaign_plan.props";

describe("CampaignPlan", () => {
  it("renders correctly", () => {
    const tree = shallow(<CampaignPlan {...CampaignPlanProps} />);
    expect(tree.debug()).toMatchSnapshot();

    tree.setState({ activeTab: "reputation_building" });
    expect(tree.debug()).toMatchSnapshot();
  });
});
