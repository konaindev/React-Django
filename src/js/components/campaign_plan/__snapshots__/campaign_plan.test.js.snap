// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CampaignPlan renders correctly 1`] = `
<div
  className="page campaign_plan-view"
>
  <div
    className="container"
  >
    <div
      className="campaign_plan-view__subnav"
    >
      <div
        className="button-group"
      >
        <button
          className="button button--selected button-undefined button-group__item"
          onClick={[Function]}
        >
          Overview
        </button>
        <button
          className="button button-undefined button-group__item"
          onClick={[Function]}
        >
          Reputation Building
        </button>
        <button
          className="button button-undefined button-group__item"
          onClick={[Function]}
        >
          Demand Creation
        </button>
        <button
          className="button button-undefined button-group__item"
          onClick={[Function]}
        >
          Leasing Enablement
        </button>
        <button
          className="button button-undefined button-group__item"
          onClick={[Function]}
        >
          Market Intelligence
        </button>
      </div>
    </div>
  </div>
  <img
    src="https://imgur.com/6pa134E.png"
  />
</div>
`;
